src/concrete/ml/onnx/ops_impl.py

```diff
@@ -893,11 +897,17 @@ def numpy_equal(
     Args:
         x (numpy.ndarray): Input tensor
         y (numpy.ndarray): Input tensor
+        lsbs_to_remove (Optional[int]): The number of the least significant bits to remove.
 
     Returns:
         Tuple[numpy.ndarray]: Output tensor
     """
 
+    # In the case of trees, x == y <=> x <= y or x < y - 1, because y is the max sum.
```

Comment:
I don't thing this is a good idea here : so we're basically implementing something very specific for trees in the numpy's implementation for equal values ? Because looking at right below, the `lambda x: x >= 0` seems very wrong to me in the general case, especially since there is not protection for not using `lsbs_to_remove` for other circuits than the ones representing a tree-based model

what I'm saying is why not just put : `rounded_comparison(x, y, lsbs_to_remove, operation=lambda x: x == 0)` ?

Reply:
We had a discussion on slack about creating new onnx nodes for tree-based models. And we agreed to not change the behavior of onnx. 

For `rounded_comparison(x, y, lsbs_to_remove, operation=lambda x: x == 0)` I responded in an other comment, it's not equivalent, you can easily do the test, you will see.

`numpy.equal(x, y)` is not equivalent to `round_bit_pattern(x - y - half) == 0`, it will probably be equivalent for truncate but not for rounding. 

We were able to find an equivalent of equal for TREES, because y is the max sum, hence, we could have x == y <=> x <= y.

Reply:
ok thanks now I get why it is not a `==`. But that means this is pretty dangerous.  Although for `execute_onnx_with_numpy` and `get_equivalent_numpy_forward_from_onnx` we can make sure models are tree-based when using the feature (through asserts, hopefully), but here we do not control anything. This `numpy_equal` op is the implementation for all equality operators used in any ONNX. This `x == y <=> x <= y or x < y - 1, because y is the max sum.` might be true for tree-based models, but it is not for all other cases we might encounter. Here, we are not able to make sure it is used for tree-based models only, it's again very a specific thing in a very generic function, without much control over it.

Reply:
> We had a discussion on slack about creating new onnx nodes for tree-based models. And we agreed to not change the behavior of onnx.

I did not agree. I still don't have an evidence of why this is risky. I still think this is the right solution to make this PR much simpler.